coderadar-graph/src/main/java/io/reflectoring/coderadar/graph/analyzer/adapter/StartAnalyzingAdapter.java

```diff
@@ -2,40 +2,23 @@
 
 import io.reflectoring.coderadar.analyzer.port.driven.StartAnalyzingPort;
 import io.reflectoring.coderadar.analyzer.port.driver.StartAnalyzingCommand;
-import io.reflectoring.coderadar.graph.analyzer.domain.AnalyzingJobEntity;
-import io.reflectoring.coderadar.graph.analyzer.repository.AnalyzingJobRepository;
 import io.reflectoring.coderadar.graph.projectadministration.project.repository.ProjectRepository;
-import java.util.Optional;
+import io.reflectoring.coderadar.projectadministration.ProjectNotFoundException;
 import org.springframework.stereotype.Service;
 
 @Service
 public class StartAnalyzingAdapter implements StartAnalyzingPort {
   private final ProjectRepository projectRepository;
-  private final AnalyzingJobRepository analyzingJobRepository;
 
-  public StartAnalyzingAdapter(
-      ProjectRepository projectRepository, AnalyzingJobRepository analyzingJobRepository) {
+  public StartAnalyzingAdapter(ProjectRepository projectRepository) {
     this.projectRepository = projectRepository;
-    this.analyzingJobRepository = analyzingJobRepository;
   }
 
   @Override
-  public Long start(StartAnalyzingCommand command, Long projectId) {
-    Optional<AnalyzingJobEntity> analyzingJobEntity =
-        analyzingJobRepository.findByProjectId(projectId);
-
-    if (analyzingJobEntity.isPresent()) {
-      analyzingJobEntity.get().setActive(true);
-      return analyzingJobRepository.save(analyzingJobEntity.get()).getId();
-    } else {
-      AnalyzingJobEntity analyzingJob = new AnalyzingJobEntity();
-      analyzingJob.setRescan(command.getRescan());
-      analyzingJob.setFrom(command.getFrom());
-      analyzingJob.setActive(true);
-
-      analyzingJob = analyzingJobRepository.save(analyzingJob);
-      projectRepository.setAnalyzingJob(projectId, analyzingJob.getId());
-      return analyzingJob.getId();
+  public void start(StartAnalyzingCommand command, Long projectId) {
+    if (!projectRepository.existsById(projectId)) {
+      throw new ProjectNotFoundException(projectId);
```

Comment:
I think this should probably be tested in the StartAnalyzingService, and not in the Adapter.

Reply:
Yes, you are probably right. In this case we'll have to do it in all adapters.